### DISCUSSION TITLE
Graphormer and Transformer vs Convolution in TransformerConv

### DISCUSSION THREAD

**Discussion by Nokimann** [pyg-team/pytorch_geometric#5204] [General]
In fact, [TransformerConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.TransformerConv) is more like to be close to the convolutional type.

Unless the `edge_index` is the full connected graph with self-loops of all nodes (= all 1s in the adjacency matrix), we cannot call it `graphormer` or `transformer` as mostly seen in NPL or Vision.

What do you think about it?

**Comment by rusty1s** [comment]
Yes, we refer here to a `SparseTransformer` variant based on sparse-connections/an `edge_index` representation. As the underlying computation is the same, I think the name is still fitting. Wouldn't you agree?
